src/object/schema_manager.c

```diff
@@ -2249,23 +2277,88 @@ sm_qualifier_name (const char *name, char *buf, int buf_size)
       return NULL;
     }
 
+  assert (buf != NULL);
+  assert (buf_size > 0);
+
   /* If the name is not a user-specified name, NULL is returned. */
   dot = strchr (name, '.');
   if (dot == NULL)
     {
       return NULL;
     }
 
-  len = STATIC_CAST (int, dot - name);
+  name_size = STATIC_CAST (int, dot - name);
+  if (name_size >= buf_size || name_size >= DB_MAX_USER_LENGTH)
+    {
+      /* exit on error */
 
-  assert (buf != NULL);
-  assert (len > 0 && len < buf_size);
-  memcpy (buf, name, len);
-  buf[len] = '\0';
+      char name_copy[DB_MAX_USER_LENGTH] = { '\0' };
+      int n_bytes = (buf_size <= DB_MAX_USER_LENGTH) ? buf_size : DB_MAX_USER_LENGTH;
+
+      if (name_size < DB_MAX_USER_LENGTH)
+	{
+	  memcpy (name_copy, name, name_size);
+	  name_copy[name_size] = '\0';
+	}
+      else
+	{
+	  memcpy (name_copy, name, (DB_MAX_USER_LENGTH - 4));
+	  memcpy (name_copy + (DB_MAX_USER_LENGTH - 4), "...", 4);
+	}
+
+      ERROR_SET_ERROR_2ARGS (error, ER_SM_MAX_LENGTH_EXCEEDED_USER, name_copy, n_bytes);
+      return NULL;
+    }
+
+  memcpy (buf, name, name_size);
+  buf[name_size] = '\0';
 
   return buf;
 }
 
+/*
+ * sm_check_qualifier_name_size() - Checks the length of the user name used as a qualifier.
+ *   return: true if there is no problem, , false otherwise.
+ *   name(in): user-specified name or object name
+ */
+bool
+sm_check_qualifier_name_size (const char *name)
+{
+  const char *dot = NULL;
+  int name_size = 0;
+  int error = NO_ERROR;
+
+  if (name == NULL || name[0] == '\0')
+    {
+      ERROR_SET_WARNING (error, ER_SM_INVALID_ARGUMENTS);
+      return NULL;
+    }
+
+  /* If the name is not a user-specified name, true is returned. */
+  dot = strchr (name, '.');
+  if (dot == NULL)
+    {
+      return true;
+    }
+
+  name_size = STATIC_CAST (int, dot - name);
+  if (name_size >= DB_MAX_USER_LENGTH)
+    {
+      /* exit on error */
+
+      char name_copy[DB_MAX_USER_LENGTH] = { '\0' };
+      int n_bytes = DB_MAX_USER_LENGTH;
+
+      memcpy (name_copy, name, (DB_MAX_USER_LENGTH - 4));
+      memcpy (name_copy + (DB_MAX_USER_LENGTH - 4), "...", 4);
+
+      ERROR_SET_ERROR_2ARGS (error, ER_SM_MAX_LENGTH_EXCEEDED_USER, name_copy, n_bytes);
```

Comment:
Is there a length limit for error messages?
I think it can be solved with just one lines like below.

ERROR_SET_ERROR_2ARGS (error, ER_SM_MAX_LENGTH_EXCEEDED_USER, name, B_MAX_USER_LENGTH);

Reply:
As you said, I removed this code because the function to check the length was run before.
Thank you.